### DISCUSSION TITLE
Why is the library git tag based ?

### DISCUSSION THREAD

**Discussion by istvandesign** [jscutlery/semver#673] [General]
Hello,
I wonder why is the source of truth git ? Why didn't the author take each package.json and parse them individually to bump the version ?

I have endless discussions with architects on why the front-end relies on git for versioning and the code is not versioned like with pom.xml/graddle...

Thanks

**Comment by edbzn** [comment]
Hi, it's a design choice, and we want to stick with it. In any case we need Git to calculate the new version by analyzing the commits history. The single source of truth is Git tags because it's easier to answer the following question "since when should we parse the git history to calculate the new version". If we introduce some fallbacks to package.json then we could have inconsistent version calculation.
